tensorflow_lite_support/python/task/audio/core/tensor_audio.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TensorAudio class."""
+
+import numpy as np
+
+from tensorflow_lite_support.python.task.audio.core.pybinds import _pywrap_audio_buffer
+from tensorflow_lite_support.python.task.audio.core import audio_record
+
+_CppAudioBuffer = _pywrap_audio_buffer.AudioBuffer
+_CppAudioFormat = _pywrap_audio_buffer.AudioFormat
+_LoadAudioBufferFromFile = _pywrap_audio_buffer.LoadAudioBufferFromFile
+
+
+class TensorAudio(object):
+  """A wrapper class to store the input audio."""
+
+  def __init__(self,
+               audio_format: _CppAudioFormat,
+               sample_count: int,
+               audio_data: _CppAudioBuffer = None,
+               is_from_file: bool = False,
+               ) -> None:
+    """Initializes the `TensorAudio` object.
+
+    Args:
+      audio_format: C++ AudioFormat object, format of the audio.
+      sample_count: int, number of samples in the audio.
+      audio_data: C++ AudioBuffer object, contains raw audio data, buffer size
+      and audio format info.
+      is_from_file: boolean, whether `audio_data` is loaded from the audio file.
+    """
+    self._format = audio_format
+    self._sample_count = sample_count
+    self._is_from_file = is_from_file
+
+    if self._is_from_file:
+      self._data = audio_data
+    else:
+      self._buffer = np.zeros(
+        [self._sample_count, self._format.channels], dtype=np.float32)
+
+  def clear(self):
+    """Clear the internal buffer and fill it with zeros."""
+    self._buffer.fill(0)
+
+  @classmethod
+  def create_from_wav_file(cls,
```

Comment:
* buffer_size should be optional. If it's not specified, buffer_size should be the full size of the WAV file. If buffer_size is specified, only load the buffer_size from the beginning of the WAV file. Please document this behavior in docstring.
* Please also update the C++ LoadAudioBufferFromFile function to follow this new behavior so that we have consistency between the two.

Reply:
@khanhlvg The example [here](https://github.com/tensorflow/tflite-support/blob/5490f096933fe0348d45c8ca8c955e1f158f8971/tensorflow_lite_support/examples/task/audio/desktop/audio_classifier_lib.cc#L77) seems to expect the buffer size as a required input for the LoadAudioBufferFromFile function. How do we determine the size of the buffer [here](https://github.com/khanhlvg/tflite-support/blob/45e15cf5973f3462095a00362cb544c16b54d5d9/tensorflow_lite_support/python/task/audio/core/tensor_audio.py#L79) in Python if the buffer size is unspecified?

Reply:
Please disregard this comment. I was thinking that we should provide an option to load the whole WAV file into the TensorAudio, but then I realized that Audio classifier still doesn't support classifying an arbitrary size audioBuffer yet, so let's keep it as how it's now.

Reply:
Okay, got it.

Reply:
Can you remove audio_data from initialization params?

Reply:
@khanhlvg How do we save the float buffer object with the `create_from_wav_file` method being a classmethod?

Reply:
Can you get the float buffer from the loaded AudioBuffer, then create a numpy array from it and call load_from_array after initializing the TensorAudio?

You can do something like this:
https://github.com/tensorflow/tflite-support/blob/5490f096933fe0348d45c8ca8c955e1f158f8971/tensorflow_lite_support/python/task/vision/core/tensor_image.py#L96